chia/_tests/core/test_merkle_set.py

```diff
@@ -24,18 +24,13 @@ async def test_basics(bt: BlockTools) -> None:
     num_blocks = 20
     blocks = bt.get_consecutive_blocks(num_blocks)
 
-    merkle_set = MerkleSet()
-    merkle_set_reverse = MerkleSet()
     coins = list(itertools.chain.from_iterable(map(lambda block: block.get_included_reward_coins(), blocks)))
 
     # excluded coin (not present in 'coins' and Merkle sets)
     excl_coin = coins.pop()
 
-    for coin in reversed(coins):
-        merkle_set_reverse.add_already_hashed(coin.name())
-
-    for coin in coins:
-        merkle_set.add_already_hashed(coin.name())
+    merkle_set_reverse = MerkleSet([coin.name() for coin in reversed(coins)])
+    merkle_set = MerkleSet([coin.name() for coin in coins])
```

Comment:
I'm not confident that the rust binding will actually support `Iterable` here, but probably `Sequence`

Reply:
the rust constructor takes a `PyList` right now, and I'm unsure if pyo3 adds any conversion magic in its binding. I would expect when you take a rust type, like `Vec` it probably does. It would probably not be too much work to change it to `PySequence`, but looking at it, it might not be a drop-in replacement. So, on a second through, I think it probably makes sense to just leave this as `List` until we need it to be more generic.